src/daft-table/src/lib.rs

```diff
@@ -173,6 +173,21 @@ impl Table {
         }
     }
 
+    pub fn add_monotonically_increasing_id(
+        &self,
+        partition_num: u64,
+        offset: u64,
+        column_name: &str,
+    ) -> DaftResult<Self> {
+        // Use the leftmost 28 bits for the partition number and the rightmost 36 bits for the row number
+        let partition_num = partition_num << 36;
```

Comment:
we already have a `UInt64Array::arange(start, end)` that you can use here.

Reply:
https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/arange.rs#L12-L23 arange operates on i64s though, would this be ok?